bentoml/_internal/frameworks/onnxmlir.py

```diff
@@ -0,0 +1,396 @@
+        context=context,
+    ) as bento_model:
+        if trust_compiled_model and model_path.endswith(".so"):
+            shutil.copyfile(model_path, bento_model.path_of(MODEL_FILENAME))
+            return bento_model
+        if ONNX_MLIR:
+            subprocess.check_output(
+                [
+                    ONNX_MLIR,
+                    *COMPILE_ARG,
+                    model_path,
+                    "-o",
+                    bento_model.path_of(MODEL_FILENAME.split(".")[0]),
+                ]
+            )
+        else:
+            # docker run --rm -it -v $(pwd):/workdir onnxmlirczar/onnx-mlir:latest -O3 --EmitLib /workdir/docs/mnist_example/mnist.onnx -o torch_mnist
+            logger.info(
+                f"Using {RELEASE_MLIR_IMAGE} to compile given model {model_path}. It might take a while if this is the first time you download {RELEASE_MLIR_IMAGE}."
+            )
+            if platform.machine() not in ["x86_64", "ppc64le", "s390x"]:
+                raise BentoMLException(
+                    f"onnxmlirczar/onnx-mlir doesn't contain a image for {platform.machine()}.\n{warning}\nRefers to https://github.com/onnx/onnx-mlir for more information."
+                )
+            model_parent_dir = os.path.dirname(model_path)
+            target_path = bento_model.path_of("/")
+            DOCKER_CMD = [
+                "run",
+                "--rm",
+                "-it",
+                "--platform",
+                UNAME_M_TO_PLATFORM_MAPPING[docker_platform],
+                "-v",
+                f"{model_parent_dir}:{DOCKER_WORKDIR}{model_parent_dir}",
+                "-v",
+                f"{target_path}:{DOCKER_OUTDIR}{target_path}",
+                "-u",
+                f"{os.geteuid()}:{os.getegid()}",
+                RELEASE_MLIR_IMAGE,
+                *COMPILE_ARG,
+                f"{DOCKER_WORKDIR}{model_path}",
+                "-o",
+                f"{DOCKER_OUTDIR}{bento_model.path_of(MODEL_FILENAME).split('.')[0]}",
+            ]
+            run_docker_cmd(DOCKER_CMD)
+
+        return bento_model
+
+
+def get_runnable(bento_model: bentoml.Model) -> t.Type[bentoml.Runnable]:
+    """
+    Private API: use :obj:`~bentoml.Model.to_runnable` instead.
+    """
+
+    class ONNXMLirRunnable(bentoml.Runnable):
+        SUPPORTED_RESOURCES = ("cpu",)  # TODO: support other accelerators
+        SUPPORTS_CPU_MULTI_THREADING = True
+
+        def __init__(self):
+            super().__init__()
+
+            self.model = load_model(bento_model)
+
+            self.entrypoints: dict[str, t.Callable[..., t.Any]] = {}
+            for method_name in bento_model.info.signatures:
+                try:
+                    self.entrypoints[method_name] = getattr(self.model, method_name)
+                except AttributeError:
+                    raise InvalidArgument(
+                        f"No method with name {method_name} found for compiled model {self.model}"
+                    )
```

Comment:
this should work with `use_multiple_entry_point` as bentoml runner also have a signature feature.